sbc_compassion/models/sponsorship_correspondence.py

```diff
@@ -172,38 +171,50 @@ def get_communication_types(self):
             ('Supporter Letter', _('Supporter Letter')),
         ]
 
+    @api.multi
     @api.depends('sponsorship_id')
     def _set_name(self):
-        if self.sponsorship_id and self.communication_type_ids:
-            self.name = self.communication_type_ids[0].name + ' (' + \
-                self.sponsorship_id.partner_codega + " - " + \
-                self.child_id.code + ')'
-        else:
-            self.name = _('New correspondence')
+        for corr in self:
```

Comment:
Choose either `correspondence` or `letter` for naming one record of `sponsorship.correspondence` and stick to that name in the whole code for consistency.